Mathlib/Algebra/Order/Monoid/Unbundled/WithTop.lean

```diff
@@ -235,6 +235,17 @@ protected theorem add_lt_add_of_lt_of_le [Preorder α] [CovariantClass α α (·
     a + c < b + d :=
   (WithTop.add_lt_add_right hc hab).trans_le <| add_le_add_left hcd _
 
+lemma addLECancellable_of_ne_top [Preorder α] [ContravariantClass α α (· + ·) (· ≤ ·)]
```

Comment:
This will allow us to write `.of_ne_top h` if the expected type is known.
```suggestion
lemma AddLECancellable.of_ne_top [Preorder α] [ContravariantClass α α (· + ·) (· ≤ ·)]
```

Reply:
This doesn't work because it's in the `WithTop` namespace